src/Microsoft.IdentityModel.Tokens/CompressionProviderFactory.cs

```diff
@@ -0,0 +1,245 @@
+        ///// Compress the value using the given algorithm
+        ///// </summary>
+        ///// <param name="algorithm">Compression algorithm</param>
+        ///// <param name="value">The value to compress</param>
+        ///// <returns></returns>
+        //public byte[] Compress(string algorithm, string value)
+        //{
+        //    if (algorithm == null)
+        //        throw LogHelper.LogArgumentNullException(algorithm);
+
+        //    if (string.IsNullOrEmpty(value))
+        //        return null;
+
+        //    if (CustomCompressionProvider != null && CustomCompressionProvider.IsSupportedAlgorithm(algorithm))
+        //    {
+        //        var compressed = CustomCompressionProvider.Compress(algorithm, value);
+        //        if (compressed == null)
+        //            throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(LogMessages.IDX10674, algorithm)));
+        //        return compressed;
+        //    }
+
+        //    if (IsSupportedAlgorithm(algorithm))
+        //        return CompressUtil(algorithm, value);
+
+        //    throw LogHelper.LogExceptionMessage(new NotSupportedException(LogHelper.FormatInvariant(LogMessages.IDX10652, algorithm)));
+        //}
+
+        //private byte[] CompressUtil(string algorithm, string value)
+        //{
+        //    if (algorithm.Equals(CompressionAlgorithm.Deflate))
+        //        return CompressionUtils.CompressWithDeflate(value);
+
+        //    return null;
+        //}
+    }
+
+    /// <summary>
+    /// Compression algorithms.
+    /// </summary>
+    public class CompressionAlgorithm
+    {
+#pragma warning disable 1591
+        public const string Deflate = "DEF";
+#pragma warning restore 1591
+    }
+
+    /// <summary>
+    /// Utility function for compression and decompression.
+    /// </summary>
+    public class CompressionUtils
+    {
+        /// <summary>
+        /// Compress the value using DEFLATE algorithm.
+        /// </summary>
+        /// <param name="value">The string to compress</param>
+        /// <returns>Compression bytes</returns>
+        public static byte[] CompressWithDeflate(string value)
+        {
+            using (MemoryStream output = new MemoryStream())
+            {
+                using (DeflateStream deflateStream = new DeflateStream(output, CompressionMode.Compress))
+                {
+                    using (StreamWriter writer = new StreamWriter(deflateStream, Encoding.UTF8))
+                    {
+                        writer.Write(value);
+                    }
+                }
+
+                return output.ToArray();
+            }
+        }
+
+        /// <summary>
+        /// Decompress the value using DEFLATE algorithm.
+        /// </summary>
+        /// <param name="value">The bytes to decompress</param>
+        /// <returns>Decompression string</returns>
+        public static string DecompressWithDeflate(byte[] value)
```

Comment:
We need to validate parameters, check for null.
Are there any other possible values that don't make sense?